src/Generation/generator.py

```diff
@@ -0,0 +1,82 @@
+import json
+import regex
+from llm.connector import connector
+
+DESCRIPTION_DIRECTORY = "Generation/Descriptions"
+FORM_FILE_NAME = "form.json"
+QUERY_FILE_NAME = "query_text"
+EXAMPLES_FILE_NAME = "examples.json"
+PREDEFINED_FILE_NAME = "predefined.json"
+FORM_FILE = DESCRIPTION_DIRECTORY + "/" + "{type_name}/" + FORM_FILE_NAME
+QUERY_FILE = DESCRIPTION_DIRECTORY + "/" + "{type_name}/" + QUERY_FILE_NAME
+EXAMPLES_FILE = DESCRIPTION_DIRECTORY + "/" + "{type_name}/" + EXAMPLES_FILE_NAME
+PREDEFINED_FILE = DESCRIPTION_DIRECTORY + "/" + "{type_name}/" + PREDEFINED_FILE_NAME
+EXAMPLES_KEY = "Examples"
+
+class Form:
+    def __init__(self, form_text: str):
+        self._form = form_text
+        self._keys = list(json.loads(self._form).keys())
+
+    # TODO: Maybe we should apply a more detailed check just to be sure
+    def FormatCheck(self, data_to_check: dict) -> bool:
```

Comment:
I've heard good things about https://pydantic.dev/, if we need further validation.

Reply:
I'll take a look. Especially because of cohere :D